shared/macros/10-ansible.jinja

```diff
@@ -1022,3 +1042,345 @@ Part of the grub2_bootloader_argument_absent template.
 {{%- endmacro -%}}
+
+
+{{#
+    Make sure that a line with a specific PAM module is present with the correct control.
+    If the line is not present, it will be included after the regex informed in the "after_match"
+    parameter. If the "after_match" parameter is empty, the line will be included at the end of
+    the file informed in the "pam_file" parameter.
+    If the line was already present, but with a different control, the control will be updated.
+    Note: If there are multiple lines matching the "group" + "module", all lines will be updated.
+    There are cases where modules are configured by multiple lines, like pam_sss.so and
+    pam_faillock.so. In some situations, these special modules may have similar lines sharing the
+    same "group" and "module". For these specific cases, carefull tests should be done to make sure
+    the module is working as expected.
+
+:param pam_file:        PAM config file.
+:param group:           PAM management group: auth, account, password or session. Also known as "type".
+:param control:         PAM control flags.
+:param module:          PAM module name.
+:param after_match:     Regex used as reference to include the line below, if necessary. Optional parameter.
+
+#}}
+{{%- macro ansible_ensure_pam_module_line(pam_file, group, control, module, after_match='') -%}}
+- name: '{{{ rule_title }}} - Check if expected PAM module line is present in {{{ pam_file }}}'
+  ansible.builtin.lineinfile:
+    path: "{{{ pam_file }}}"
+    regexp: ^\s*{{{ group }}}\s+{{{ control }}}\s+{{{ module }}}\s*.*
+    state: absent
+  check_mode: yes
+  changed_when: false
+  register: result_pam_line_present
+
+- name: '{{{ rule_title }}} - Include or update the PAM module line in {{{ pam_file }}}'
+  block:
+    - name: '{{{ rule_title }}} - Check if required PAM module line is present in {{{ pam_file }}} with different control'
+      ansible.builtin.lineinfile:
+        path: "{{{ pam_file }}}"
+        regexp: ^\s*{{{ group }}}\s+.*\s+{{{ module }}}\s*
+        state: absent
+      check_mode: yes
+      changed_when: false
+      register: result_pam_line_other_control_present
+
+    - name: '{{{ rule_title }}} - Ensure the correct control for the required PAM module line in {{{ pam_file }}}'
+      ansible.builtin.replace:
+        dest: "{{{ pam_file }}}"
+        regexp: '^(\s*{{{ group }}}\s+).*(\b{{{ module }}}.*)'
+        replace: '\1{{{ control }}} \2'
+      when:
+        - result_pam_line_other_control_present.found > 0
+
+    - name: '{{{ rule_title }}} - Ensure the required PAM module line is included in {{{ pam_file }}}'
+      ansible.builtin.lineinfile:
+        dest: "{{{ pam_file }}}"
+        {{%- if after_match != '' %}}
+        insertafter: {{{ after_match }}}
+        {{%- endif %}}
+        line: {{{ group }}}    {{{ control }}}    {{{ module }}}
+      when:
+        - result_pam_line_other_control_present.found == 0
+  when:
+    - result_pam_line_present.found == 0
```

Comment:
The `ansible_ensure_pam_module_line` macro is not called directly. It is recommended to use the `ansible_ensure_pam_module_configuration` instead and this macro already test the file existence. In this case, it failed because an Ansible variable was incorrect, as mentioned in another comment. This will be fixed in the next commit. However, I see benefits in this suggested condition, but not replacing the existing one. I will include it.